docs/content/docs/dev/datastream/how_to_migrate_from_dataset_to_datastream.md

```diff
@@ -0,0 +1,660 @@
+{{< tabs projectfunc>}}
+{{< tab "DataSet">}}
+```java
+DataSet<Tuple3<Integer, Double, String>> input = // [...]
+DataSet<Tuple2<String, Integer>> out = input.project(2,0);
+```
+{{< /tab >}}
+{{< tab "DataStream">}}
+```java
+DataStream<Tuple3<Integer, Double, String>> input = // [...]
+DataStream<Tuple2<String, Integer>> out = input.map(value -> Tuple2.of(value.f2, value.f0));
+```
+{{< /tab >}}
+{{< /tabs>}}
+
+#### Distinct
+
+{{< tabs distinctfunc>}}
+{{< tab "DataSet">}}
+```java
+DataSet<Integer> input = // [...]
+DataSet<Integer> output = input.distinct();
+```
+{{< /tab >}}
+{{< tab "DataStream">}}
+```java
+DataStream<Integer> input = // [...]
+DataStream<Integer> output = input
+        .keyBy(value -> value)
+        .reduce((value1, value2) -> value1);
+```
+{{< /tab >}}
+{{< /tabs>}}
+
+#### Hash-Partition
+
+{{< tabs hashpartitionfunc>}}
+{{< tab "DataSet">}}
+```java
+DataSet<Tuple2<String, Integer>> input = // [...]
+DataSet<Tuple2<String, Integer>> output = input.partitionByHash(value -> value.f0);
+```
+{{< /tab >}}
+{{< tab "DataStream">}}
+```java
+DataStream<Tuple2<String, Integer>> input = // [...]
+DataStream<Tuple2<String, Integer>> output =
+        input.partitionCustom(
+        (key, numSubpartition) ->
+        // partition by the hashcode of key
+        key.hashCode() % numSubpartition, value -> value.f0);
+```
+{{< /tab >}}
+{{< /tabs>}}
+
+#### Reduce on Full DataSet
+
+{{< tabs reducefullfunc>}}
+{{< tab "DataSet">}}
+```java
+DataSet<String> input = // [...]
+DataSet<String> output = input.reduce(new ReduceFunction(){
+        // implement user-defined reduce logic
+        });
+```
+{{< /tab >}}
+{{< tab "DataStream">}}
+```java
+DataStream<String> input = // [...]
+DataStream<String> output = input.keyBy(value -> "KEY")
+        .reduce(new ReduceFunction(){
+        // implement user-defined reduce logic
+        });
+```
+{{< /tab >}}
+{{< /tabs>}}
+
+
+#### Aggregate on Full DataSet
+
+{{< tabs aggregatefullfunc>}}
+{{< tab "DataSet">}}
+```java
+DataSet<Tuple2<Integer, Integer>> input = // [...]
+DataSet<Tuple2<Integer, Integer>> output = input
+        // compute sum of the second field
+        // .aggregate(SUM, 1);
+        // compute minimum value of the second field
+        // .aggregate(MIN, 1);
+        // compute maximum value of the second field
+        // .aggregate(MAX, 1);
+```
+{{< /tab >}}
+{{< tab "DataStream">}}
+```java
+DataStream<Tuple2<Integer, Integer>> input = // [...]
+        DataStream<Tuple2<Integer, Integer>> output = input
+        .keyBy(value -> "KEY")
+        // compute sum of the second field
+        // .sum(1);
+        // compute minimum value of the second field
+        // .min(1);
+        // compute maximum value of the second field
+        // .max(1);
+```
+{{< /tab >}}
+{{< /tabs>}}
+
+### Additional Computation and Shuffle Cost
+
+In the third type of operators, the complete records of the DataSet or the records of individual subtask needs to be collected 
```

Comment:
Agreed. For the third category, it differs from other categories because it will be migrated to DataStream APIs with different semantic and different processing behavior.